### ISSUE TITLE
Popup is under going some changes

### ISSUE THREAD

**Issue by ThePrimeagen** [ThePrimeagen/harpoon#402]
Highlight groups and popup window has been reworked.

we are going to drop the plenary deps and just use the nvim_open_win function.  This also means that i need to expose some border options.

as for highlight groups, there is a new event that emits out window and buf.  but i am unsure what else is needed for highlight groups

**Comment by ThePrimeagen** [comment]
Yes, I would like to drive as much as possible settings through the function itself, not some distant settings

**Comment by Arisa-Snowbell** [comment]
Yeah, the window looks completely different, the most annoying thing is background color is wrong because it doesn't use the highlight rn 
![image](https://github.com/ThePrimeagen/harpoon/assets/22961592/65910d8b-8eed-455b-83eb-4bbc0944d6da)

**Comment by patricio-ferraggi-deel** [comment]
yeah, reverted a few commits back because I like my highlight groups 😆
